### DISCUSSION TITLE
ytdlp 音源调用时顯示 Promise.any 錯誤

### DISCUSSION THREAD

**Discussion by EdisonJwa** [1715173329/UnblockNeteaseMusic#324] [Q&A] [answered]
报错代码如下（不是网络问题，手动调用时可以正常返回数据）

```
ERROR: (hook) Promise.any is not a function
    TypeError: Promise.any is not a function
        at match (/root/UnblockNeteaseMusic-fork/src/provider/match.js:64:34)
        at processTicksAndRejections (internal/process/task_queues.js:95:5)
        at async Promise.all (index 0)
INFO: (spawn) running yt-dlp -f 140 --dump-json ytsearch1:하나뿐인 사람 - K.Will
(node:3338) UnhandledPromiseRejectionWarning: undefined
(Use `node --trace-warnings ...` to show where the warning was created)
(node:3338) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:3338) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
INFO: (spawn) running yt-dlp -f 140 --dump-json https://www.youtube.com/watch?v=ug1kMvF1pKY
WARN: (request) The request timed out, or the requester didn't handle the response.
```

**Comment by pan93412** [comment, marked as answer]
> 报错代码如下（不是网络问题，手动调用时可以正常返回数据）
> 
> ```
> ERROR: (hook) Promise.any is not a function
>     TypeError: Promise.any is not a function
>         at match (/root/UnblockNeteaseMusic-fork/src/provider/match.js:64:34)
>         at processTicksAndRejections (internal/process/task_queues.js:95:5)
>         at async Promise.all (index 0)
> INFO: (spawn) running yt-dlp -f 140 --dump-json ytsearch1:하나뿐인 사람 - K.Will
> (node:3338) UnhandledPromiseRejectionWarning: undefined
> (Use `node --trace-warnings ...` to show where the warning was created)
> (node:3338) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
> (node:3338) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> INFO: (spawn) running yt-dlp -f 140 --dump-json https://www.youtube.com/watch?v=ug1kMvF1pKY
> WARN: (request) The request timed out, or the requester didn't handle the response.
> ```

升級成 Node.js 16 以上版本，或者是改用 precompiled 版本（直接啟動根目錄的 `app.js`）。

**Comment by EdisonJwa** [comment]
谢谢～
升级至16版本后解决了上述问题，但是就很怪 网易云只能播放十秒

```
INFO: (spawn) running youtube-dl -f 140 --dump-json ytsearch1:恋愛ING - 五月天
INFO: (spawn) running youtube-dl -f 140 --dump-json https://www.youtube.com/watch?v=iSkRGgYSQfY
INFO: (provider/match) Replaced: [27965305] 恋愛ING
    audioId: 27965305
    songName: "恋愛ING"
    url: "__REMOVED__"
```

![bd348ffb57d50a17539f17ada6d81ce1.png](https://i.yecdn.com/images/2021/11/14/bd348ffb57d50a17539f17ada6d81ce1.png)

**Reply by pan93412** [reply]
這是因為網易雲音樂的 Mac 版不能很好的支援 YouTube 音源所致。你可以先用網頁版，或者是等 #190 問題解決。
